### PR TITLE
HikariCP documentation improvements

### PR DESCRIPTION
**Pull request by Sovietaced**
### Summary

HikariCP documentation improvements

### Description

I tried setting up the driver and ran into some issues following the docs. I don't know how HikariCP works that well so it took me a couple hours to understand what was intended. 

What threw me off was that this [file](https://github.com/awslabs/aws-advanced-jdbc-wrapper/blob/main/docs/using-the-jdbc-driver/sample-code/HikariSample.java#L52) simply doesn't compile for a few reasons
* Its missing imports
* Its calling a method on a `HikariDataSource` that simply doesn't exist. 

I have updated it to configure the target data source properties correctly so they will be correctly configured on the AWS wrapper data source using [reflection](https://github.com/openbouquet/HikariCP/blob/master/src/main/java/com/zaxxer/hikari/util/PropertyElf.java).

### CODE REVIEW

File: docs/using-the-jdbc-driver/sample-code/HikariSample.java [hunk 1]
```diff
@@ -49,7 +50,8 @@ public static void main(String[] args) throws SQLException {
     targetDataSourceProps.setProperty("serverPropertyName", "db-identifier.cluster-XYZ.us-east-2.rds.amazonaws.com");
     targetDataSourceProps.setProperty("databasePropertyName", "postgres");
     targetDataSourceProps.setProperty("portPropertyName", "5432");
-    ds.setTargetDataSourceProperties(targetDataSourceProps);
+    
+    ds.addDataSourceProperty("targetDataSourceProps", targetDataSourceProps);
```
Comment (sergiyvamz) [inline docs/using-the-jdbc-driver/sample-code/HikariSample.java:54]: Property name should be "targetDataSourceProperties"

File: docs/using-the-jdbc-driver/DataSource.md [hunk 2]
```diff
@@ -77,6 +77,8 @@ To use the AWS JDBC Driver with a connection pool, you must:
    targetDataSourceProps.setProperty("serverName", "db-identifier.cluster-XYZ.us-east-2.rds.amazonaws.com");
    targetDataSourceProps.setProperty("databaseName", "postgres");
    targetDataSourceProps.setProperty("port", "5432");
+   
+   ds.addDataSourceProperty("targetDataSourceProps", targetDataSourceProps);
```
Comment (sergiyvamz) [inline docs/using-the-jdbc-driver/DataSource.md:81]: Property name should be "targetDataSourceProperties"

File: docs/using-the-jdbc-driver/sample-code/HikariSample.java [hunk 3]
```diff
@@ -49,7 +50,8 @@ public static void main(String[] args) throws SQLException {
     targetDataSourceProps.setProperty("serverPropertyName", "db-identifier.cluster-XYZ.us-east-2.rds.amazonaws.com");
     targetDataSourceProps.setProperty("databasePropertyName", "postgres");
     targetDataSourceProps.setProperty("portPropertyName", "5432");
```
Comment (karenc-bq) [inline docs/using-the-jdbc-driver/sample-code/HikariSample.java:52]: These need to be changed as well
```suggestion
    targetDataSourceProps.setProperty("serverName", "db-identifier.cluster-XYZ.us-east-2.rds.amazonaws.com");
    targetDataSourceProps.setProperty("databaseName", "postgres");
    targetDataSourceProps.setProperty("portNumber", "5432");
```

Similarly the code above from lines 41-43 should be changed to:
```
    ds.addDataSourceProperty("databasePropertyName", "databaseName");
    ds.addDataSourceProperty("portPropertyName", "portNumber");
    ds.addDataSourceProperty("serverPropertyName", "serverName");
```

**Comment by karenc-bq** [comment]
Thank you for catching and fixing these mistakes 👍

File: docs/using-the-jdbc-driver/DataSource.md (hunk 2 above)
Comment (Sovietaced) [reply docs/using-the-jdbc-driver/DataSource.md]: Addressed in https://github.com/awslabs/aws-advanced-jdbc-wrapper/pull/213/commits/f23fd3aa62a806dc2b603934a3df26278e294617

File: docs/using-the-jdbc-driver/sample-code/HikariSample.java (hunk 1 above)
Comment (Sovietaced) [reply docs/using-the-jdbc-driver/sample-code/HikariSample.java]: Addressed in https://github.com/awslabs/aws-advanced-jdbc-wrapper/pull/213/commits/9518e5c47a38815df720af2edbf56f397be861a3

**Review by sergiyvamz — APPROVED** [review]

**Pull request merged**